src/it/gwt-jsonix-execution-test/src/test/java/trial/execution/mapper/JsUtilsTest.java

```diff
@@ -76,16 +76,49 @@ public void testToJsArrayLike() {
     }
 
     public void testToAttributesMap() {
-        Map<QName, String> object = getObjectToMap();
+        Object object = getNativeAttributesMap();
         final Map<QName, String> retrieved = JsUtils.toAttributesMap(object);
         assertNotNull(retrieved);
-        // TODO {gcardosi} finish this test
+        assertEquals(2, retrieved.size());
+        final List<QName> list = new ArrayList<>(retrieved.keySet());
+        assertEquals("1", retrieved.get(list.get(0)));
+        assertEquals("2", retrieved.get(list.get(1)));
+    }
+
+    private static native Object getNativeAttributesMap() /*-{
+        var jso = {};
+        jso["one"] = "1";
+        jso["two"] = "2";
+        return jso;
+    }-*/;
+
+    public void testToAttributesMapNull() {
+        Map<QName, String> object = null;
+        final Map<QName, String> retrieved = JsUtils.toAttributesMap(object);
+        assertNotNull(retrieved);
+        assertEquals(0, retrieved.size());
     }
 
     public void testFromAttributesMap() {
-        // TODO (?) {gcardosi}
+        final Map<QName, String> object = new HashMap<>();
+        final QName qname1 = new QName("", "one", "");
+        final QName qname2 = new QName("", "two", "");
+        object.put(qname1, "1");
+        object.put(qname2, "2");
+
+        final Object retrieved = JsUtils.fromAttributesMap(object);
+        assertNotNull(retrieved);
+
+        assertNativeKeyIsMappedToValue(qname1.toString(), "1", retrieved);
+        assertNativeKeyIsMappedToValue(qname2.toString(), "2", retrieved);
```

Comment:
`assertNativeKeyIsMappedToValue` just returns boolean, I think wee need to assert result of `assertNativeKeyIsMappedToValue`

Reply:
That was my intention; but it somehow got forgotten. Thank-you.

Reply:
Fixed in new commit.